code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/EnvironmentCreateValidationFilter.java

```diff
@@ -29,7 +29,7 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
         }
 
         Environment existingEnv = objectManager.findOne(Environment.class, ENVIRONMENT.NAME, env.getName(),
```

Comment:
When looking up resource in the API layer it is far better to use `io.github.ibuildthecloud.gdapi.request.resource.ResourceManager` because that does proper auth checking